File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        ans = []
        check = Counter(nums2)
        for i in range(len(nums1)):
            if check[nums1[i]] > 0:
                ans.append(nums1[i])
                check[nums1[i]] -= 1

        return ans
        

File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.go
func intersect(nums1 []int, nums2 []int) []int {
    ans := []int{}
    cnt := make(map[int]int)

    for _, num := range nums1 {
        cnt[num]++
    }

    for _, num := range nums2 {
        if cnt[num] > 0 {
            ans = append(ans, num)
            cnt[num]--
        }
    }
    return ans
}
